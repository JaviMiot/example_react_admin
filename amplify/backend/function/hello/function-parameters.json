{
  "lambdaLayers": [],
  "environmentVariableList": [
    {
      "cloudFormationParameterName": "url",
      "environmentVariableName": "url"
    }
  ]
}