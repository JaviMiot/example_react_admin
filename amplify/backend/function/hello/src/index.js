const axios = require('axios');

/**
 * @type {import('@types/aws-lambda').APIGatewayProxyHandler}
 */

const urlHello = process.env.url;

exports.handler = async (event) => {
  console.log(`EVENT: ${JSON.stringify(event)}`);

  let response = {
    statusCode: 200,
    headers: {
      'Access-Control-Allow-Origin': '*',
      'Access-Control-Allow-Headers': '*',
    },
    body: '',
  };

  try {
    const responseMsg = await axios.get(urlHello);
    response.body = JSON.stringify(responseMsg.data.body);
  } catch (error) {
    response.statusCode = 500;
    response.body = JSON.stringify(error);
  }
  return response;
};
