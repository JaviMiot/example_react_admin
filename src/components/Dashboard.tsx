import React, { useState } from 'react';
import { Card, CardHeader } from '@mui/material';
import { API } from 'aws-amplify';
import { Title } from 'react-admin';
interface User {
  id: string | number;
  name: string;
}

const initUser: User = {
  id: '',
  name: '',
};

const userApi = 'user';
const pathUser = '/user';

const helloApi = 'hello';
const pathHello = '/hello';

const Dashboard = () => {
  const [user, setUser] = useState<User>(initUser);
  const [input, setInput] = useState('');

  const getUser = () => {
    API.get(userApi, `${pathUser}/${input}`, {})
      .then((response) => setUser({ ...response }))
      .catch((er) => console.log(er));
  };

  const getHello = () => {
    API.get(helloApi, pathHello, {})
      .then((response) => console.log(response))
      .catch((er) => console.log(er));
  };

  const getHelloLayer = () => {
    API.get('layerapi', '/layerapi', {})
      .then((response) => console.log(response))
      .catch((er) => console.log(er));
  };

  return (
    <Card>
      <CardHeader title={process.env.REACT_APP_TITLE} />
      <Title title="React Admin" />
      <div>
        <span>id: {user.id}</span>
        <br />
        <span>user: {user.name}</span>
      </div>
      <input type="text" onChange={(e) => setInput(e.target.value)} />
      <button onClick={getUser}>Get</button>
      <button onClick={getHello}>hello</button>
      <button onClick={getHelloLayer}>helloLayer</button>
      <p>
        Lorem ipsum dolor sit amet consectetur adipisicing elit. Neque harum
        amet repellendus ullam dignissimos reiciendis quisquam ducimus a! Quod
        nisi animi eos numquam ea. Deleniti voluptatum harum incidunt iure
        numquam.
      </p>
    </Card>
  );
};

export default Dashboard;
