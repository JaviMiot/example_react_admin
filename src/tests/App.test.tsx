import React from 'react';
import { render, screen } from '@testing-library/react';
import { Hello } from '../components/hello';
import { create, act } from 'react-test-renderer';

describe('<Hello/>', () => {
  test('renders learn react link', () => {
    const h = render(<Hello />);
    expect(screen.getByText(/Hello/)).toBeInTheDocument();
  });
});

describe('<Hello/> snapshop', () => {
  test('get snapshop', () => {
    const hello = create(<Hello />);
    expect(hello.toJSON()).toMatchSnapshot();
  });
});
