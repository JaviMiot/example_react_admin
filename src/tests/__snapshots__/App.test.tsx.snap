// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Hello/> snapshop get snapshop 1`] = `
<div>
  Hello world
</div>
`;
